nixos/modules/services/blockchain/nimbus/beacon-node.nix

```diff
@@ -0,0 +1,236 @@
+          description = mdDoc ''
+            Path of file with 32 bytes long JWT secret for Auth RPC endpoint.
+            Can be generated using 'openssl rand -hex 32'.
+          '';
+        };
+
+        subAllSubnets = mkOption {
+          type = types.bool;
+          default = false;
+          description = mdDoc "Subscribe to all attestation subnet topics.";
+        };
+
+        doppelganger = mkOption {
+          type = types.bool;
+          default = true;
+          description = mdDoc ''
+            Protection against slashing due to double-voting.
+            Means you will miss two attestations when restarting.
+          '';
+        };
+
+        suggestedFeeRecipient = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          description = mdDoc ''
+            Wallet address where transaction fee tips - priority fees,
+            unburnt portion of gas fees - will be sent.
+          '';
+        };
+
+        listenPort = mkOption {
+          type = types.port;
+          default = 9000;
+          description = mdDoc "Listen port for libp2p protocol.";
+        };
+
+        discoverPort = mkOption {
+          type = types.port;
+          default = 9000;
+          description = mdDoc "Listen port for libp2p protocol.";
+        };
+
+        nat = mkOption {
+          type = types.str;
+          default = "any";
+          example = "extip:12.34.56.78";
+          description = literalMD ''
+            Method for determining public address. (any, none, upnp, pmp, extip:<IP>)
+          '';
+        };
+
+        metrics = {
+          enable = lib.mkEnableOption (mdDoc "Nimbus beacon node metrics endpoint");
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Metrics address for beacon node.";
+          };
+          port = mkOption {
+            type = types.port;
+            default = 9100;
+            description = mdDoc "Metrics port for beacon node.";
+          };
+        };
+
+        rest = {
+          enable = lib.mkEnableOption (mdDoc "Nimbus beacon node REST API");
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Listening address of the REST API server.";
+          };
+
+          port = mkOption {
+            type = types.port;
+            default = 5052;
+            description = mdDoc "Port for the REST API server.";
+          };
+        };
+
+        extraArgs = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          example = ["--num-threads=1" "--graffiti=1337_h4x0r"];
+          description = mdDoc "Additional arguments passed to node.";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = length cfg.web3Urls > 0;
+        message = "Nimbus beacon node requires at least one Web3 URL to work!"; }
```

Comment:
This should mention the settings path.

Reply:
What do you mean "settings path"? Do you mean the `description` field of the `web3Urls` config? If so then I agree.

Reply:
```suggestion
        message = "Nimbus beacon node requires at least one Web3 URL in services.nimbus-beacon-node.web3Urls configured to work!"; }
```

Reply:
Oooh, that's what you meant, sure.